spec/support/models/post.cr

```diff
@@ -8,8 +8,11 @@ class Post < BaseModel
     column title : String
     column published_at : Time?
     has_many comments : Comment
+    has_many nice_comments : Comment,
+      query: Comment::BaseQuery.new.body(Comment::NICE_COMMENT_BODY)
     has_many taggings : Tagging
     has_many tags : Tag, through: [:taggings, :tag]
+    has_many funny_tags : Tag, through: [:taggings, :tag], query: Tag::BaseQuery.new.name(Tag::FUNNY_TAG_NAME)
```

Comment:
I don't think has many through will work as expected if the "through" has a query provided...

Reply:
That's what I was thinking too. I'd be ok with saying that the `through` can't have a `query`. Or at least to start.